GENERATING_DOCS.md

```diff
@@ -0,0 +1,44 @@
+# Generating Docs with Typedoc
+[Typedoc](http://typedoc.org/) is a documentation generator for TypeScript projects.
+
+Refer to `typedoc.json` for build configuration.
+
+1. `npm install`
+2. `npm run docs ./src/lib`
+
+TypeDoc will leverage the typings file to generate docs. The output will be located in `/docs/autogen/output.json`.
+
+
+## Generating docs for one component
+
+```
+npm run docs ./src/lib/button
+```
+
+This will produces typings errors because the specified path does not have a typings file.
+
+## Example Usage
+
+### Hiding methods in a class
```

Comment:
Rather than hiding the methods with `@hidden`, I'd rather just ignore them when we're processing the json for the docs site. The vast majority of methods and properties won't be public, so using `@hidden` would add a lot of noise to the source.